pkg/controller/clusterlogging/clusterlogging_controller.go

```diff
@@ -50,9 +55,41 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	/*************** TEST TEST TEST ***************
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return handleConfigMap(e.MetaNew) },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return handleConfigMap(e.Meta) },
+		CreateFunc:  func(e event.CreateEvent) bool { return handleConfigMap(e.Meta) },
+		GenericFunc: func(e event.GenericEvent) bool { return handleConfigMap(e.Meta) },
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+	*************** TEST TEST TEST ***************/
+
+	// Watch for changes to the proxy resource.
+	err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{})
```

Comment:
so one thing to note with doing this, i believe when we get to `Reconcile` any proxy config changes will push **that** event - so we may fail to get the clusterlogging instance.

I'm not sure if we can do an `EnqueueRequestForOwner` in this case to bypass that.
But in the case where we do get a proxy event change, we don't want to requeue it at the end of a successful run.

Reply:
ok - so how do other operators deal with this?  It seems that other operators that need to respect proxy settings would have to deal with this (unless once again logging is the pioneer that gets the arrows . . .)

Reply:
I'm learning from cluster-network-operator.  The operator has separated reconciler for each watched target(?).  In this PR, I piggybacked the proxyconfig watches in the clusterlogging reconciler...  Do you think we have to have a separate reconciler as cluster-network-operator does???
https://github.com/openshift/cluster-network-operator/blob/master/pkg/controller/add_networkconfig.go

Reply:
And by commenting out the cluster configmap watch, the e2e test passed.

Reply:
> cluster configmap watch
just to clarify and prevent confusion, this isn't a configmap -- its a non-namespaced object of type `config`

Reply:
> Do you think we have to have a separate reconciler as cluster-network-operator does?

I think this is the path we want to take as well, yes.

Reply:
Thanks!  I'm trying it and my first cut causes a strange problem. :)  If I update cluster proxy spec/status, it restarts all the pods including elasticsearch and kibana...  Obviously, there are lots more to learn...  But I'm glad I got something to pursue.